### PR TITLE
Improve tracing for to device messages

### PR DESCRIPTION
**Pull request by erikjohnston**
This is to try and help track down to device messages that are getting delayed or lost.

### CODE REVIEW

File: synapse/handlers/devicemessage.py [hunk 1]
```diff
@@ -221,17 +227,15 @@ async def send_device_message(
...
+            log_kv({"destination": destination})
+            remote_edu_contents[destination] = {
+                "messages": messages,
+                "sender": sender_user_id,
+                "type": message_type,
+                "message_id": message_id,
+                "org.matrix.opentracing_context": json_encoder.encode(context),
+            }
 
-        log_kv({"local_messages": local_messages})
```
Comment (anoadragon453) [inline synapse/handlers/devicemessage.py:232]: Looks like the new `log_kv({user_id, device_id})` above is meant to be the replacement for this `log_kv` and the one below. Is that correct?

File: synapse/logging/opentracing.py [hunk 2]
```diff
@@ -259,6 +259,14 @@ def report_span(self, span):
 logger = logging.getLogger(__name__)
 
 
+class SynapseTags:
+    # The message ID of any to_device message processed
+    TO_DEVICE_MESSAGE_ID = "to_device.message_id"
+
+    # Whether the sync response has new data to be returned to the client.
+    SYNC_RESULT = "sync.new_data"
```
Comment (anoadragon453) [inline synapse/logging/opentracing.py:267]: Is the purpose of this to check whether a `to_device` message came down `/sync` for a user? If so, then won't events other than `to_device` messages lower the signal-to-noise ratio?

File: synapse/handlers/typing.py [hunk 3]
```diff
@@ -86,6 +89,7 @@ def _reset(self) -> None:
         self._member_last_federation_poke = {}
         self.wheel_timer = WheelTimer(bucket_size=5000)
 
+    @wrap_as_background_process("typing.run_as_background_process")
```
Comment (anoadragon453) [inline synapse/handlers/typing.py:92]: Should this be `"typing._handle_timeouts"` or similar instead?

**Review by anoadragon453 — COMMENTED** [review]
Overall looks good, and hooray for more tracing :tada: 

I just had a couple questions below.

File: synapse/logging/opentracing.py (hunk 2 above)
Comment (erikjohnston) [reply synapse/logging/opentracing.py:267]: Ish, I was more interested in signalling whether we'd found anything or not. I might go in and add per-section flags later, but I'm a bit wary of adding toooo much at once

File: synapse/handlers/typing.py (hunk 3 above)
Comment (erikjohnston) [reply synapse/handlers/typing.py:92]: ... yes

File: synapse/handlers/devicemessage.py (hunk 1 above)
Comment (erikjohnston) [reply synapse/handlers/devicemessage.py:232]: Yup. This log was a bit useless as it gave the entire message, which then got truncated by jaeger, so you couldn't really see what was going on

**Review by anoadragon453 — APPROVED** [review]
lgtm

**Pull request merged**